src/nvim/screen.c

```diff
@@ -3911,14 +3911,6 @@ win_line (
      * At end of the text line.
      */
     if (c == NUL) {
-      if (eol_hl_off > 0 && vcol - eol_hl_off == (long)wp->w_virtcol
```

Comment:
I must admit that I do not understand the intended purpose of this code block. This was added in https://github.com/vim/vim/commit/a443af88a48d74b56ea9aa2aa486f52f340bbf77 (ftp://ftp.vim.org/pub/vim/patches/7.1/7.1.152). Unfortunately, neither of those provide very much context for the change.

I was unable to find any problematic interactions between hls and cursorcolumn (mentioned in the patch diff) that occur when this block is removed.